src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -993,8 +993,15 @@ public function findDefinition($id)
     {
         $id = $this->normalizeId($id);
 
+        $seen = array();
         while (isset($this->aliasDefinitions[$id])) {
             $id = (string) $this->aliasDefinitions[$id];
+
+            if (isset($seen[$id])) {
+                throw new ServiceCircularReferenceException($id, array_keys($seen));
```

Comment:
The message should be improved see the CheckCircularReferencePass

Reply:
Sorry was on mobile :)
This is what I mean:
```php
if (isset($seen[$id])) {
    $ids = array_slice($seen, array_search($id, $seen));
    $ids[] = $id;

    throw new ServiceCircularReferenceException($id, $seen);
}

$seen[$id] = $id;
```

Reply:
I see the idea. Tho, we can't use `array_slice` with string keys in an array, we'll have to use 2 of them.